ydb/core/kqp/host/kqp_host.cpp

```diff
@@ -1152,7 +1166,8 @@ class TKqpHost : public IKqpHost {
     }
 
 private:
-    TVector<TExprNode::TPtr> CompileQuery(const TKqpQueryRef& query, bool isSql, TExprContext& ctx, TMaybe<TSqlVersion>& sqlVersion, TKqpTranslationSettingsBuilder& settingsBuilder) const
+    TVector<TExprNode::TPtr> CompileQuery(const TKqpQueryRef& query, bool isSql, TExprContext& ctx, TMaybe<TSqlVersion>& sqlVersion,
+        TKqpTranslationSettingsBuilder& settingsBuilder, bool& keepInCache) const
```

Comment:
It is hard to grasp whether keepInCache is a parameter or a function result. Can we pls move it to function result, if possible? something along the lines 
```
struct TCompileQueryResult {
    TVector<TExprNode::TPtr>AstRoots;
    bool keepInCache;
};

TCompileQueryResult CompileQuery(...)
```